src/crypto/transform.py

```diff
@@ -1,69 +1,135 @@
+from crypto.utils.helpers import service_credentials
 from crypto.utils.constants import (
     BUCKET,
     CLOUD_STORAGE,
     BIGQUERY,
+    PROJECT_ID,
     DAY_DIM,
     MONTH_DIM,
-    INITIAL_LOAD
+    DATE_DIM,
+    INITIAL_LOAD,
+    NAME_DIM,
 )
+from crypto.utils.setup import DayDim, MonthDim, DateDim, NameDim
 
 
 class Transform:
 
     def __init__(self, storage_client: CSClient, bq_client: BQClient, bucket_name: str) -> None:
         self.storage_client = storage_client
         self.bq_client = bq_client
-        self.bucket_name = bucket_name
+        self._bucket = self.storage_client.bucket(bucket_name=bucket_name)
 
+    @property
     def bucket(self) -> Bucket:
-        return self.storage_client.bucket(bucket_name=self.bucket_name)
+        return self._bucket
     
     def blob(self, blob_name: str) -> Blob:
-        return self.bucket().blob(blob_name=blob_name)
+        return self.bucket.blob(blob_name=blob_name)
 
     def read_blob(self) -> dict:
-        blob_name = f"{datetime.today().strftime('%Y-%m-%d')}.json"
+        blob_name = f"{datetime.today().strftime('%Y-%m-%d')}.json" # for testing
         blob = self.blob(blob_name=blob_name)
         if blob.exists():
             with blob.open("r") as file:
                 crypto_data = json.loads(file.read())
             return crypto_data
         return {}
     
-    @time_period
-    def day_dim_rows(self) -> list:
-        return list(calendar.day_name)
+    def load_table(self, table_id: str, rows: list) -> None:
+        table = self.bq_client.get_table(table=table_id)
+
+        job_config = LoadJobConfig(
+            schema=table.schema
+        )
+
+        job = self.bq_client.load_table_from_json(
+            json_rows=rows,
+            destination=table_id,
+            project=PROJECT_ID,
+            job_config=job_config
+        )
+
+        job.result()
     
-    @time_period
+    def day_dim_rows(self) -> list:
+        rows = []
+        for key, day in enumerate(calendar.day_name, start=1):
+            row = DayDim(day_key=key, name=day)._asdict()
+            rows.append(row)
+        return rows
+
     def month_dim_rows(self) -> list:
-        return list(calendar.month_name)[1:]
+        rows = []
+        for key, month in enumerate(list(calendar.month_name)[1:], start=1):
+            row = MonthDim(month_key=key, name=month)._asdict()
+            rows.append(row)
+        return rows
     
-    def _date_dim_row(self) -> list:
-        pass
+    def date_dim_row(self) -> list:
+        d = datetime.today()
+        row = DateDim(
+            date_key=f"{d:%Y-%m-%d}",
+            year=d.year,
+            month_key=d.month,
+            day=d.day,
+            day_key=d.isoweekday(),
+            week_number=d.isocalendar().week,
+            week_end=d.fromisocalendar(d.year, d.isocalendar().week, 7).strftime('%Y-%m-%d'),
+            month_end=(d + relativedelta(day=31)).strftime("%Y-%m-%d")
+        )._asdict()
```

Comment:
funny how the Standard Library is not always consistent: this method should probably be "public" :)